File: Weapon/GunshotWeapon/Automatic.py
from GunshotWeapon import GunshotWeapon


class Automatic(GunshotWeapon):
    __rateFire = None

    @property
    def rate_fire(self):
        return self.__rateFire

    @rate_fire.setter
    def rate_fire(self, value: int):
        self.__rateFire = value

File: Weapon/GunshotWeapon/Automatic.php
<?php


namespace Weapon;


class Automatic extends GunshotWeapon
{
    private $rateFire;

    final public function getRateFire(){
        return $this->rateFire;
    }
    final public function setRateFire(float $rateFire){
        $this->rateFire = $rateFire;
    }

}